src/core/event_engine/effect_invocation.rs

```diff
@@ -0,0 +1,21 @@
+use crate::core::event_engine::Effect;
+
+/// Effect invocation trait.
+///
+/// Invocation is an intention to run an effect. Effect dispatcher uses intents
+/// to schedule actual effect invocation.
+pub(crate) trait EffectInvocation {
```

Comment:
Do we need that trait?

Reply:
Yup. 
This is the base trait for other types of state machine `effect invocation` (right now only _subscribe_, but there will be more).